RFS/src/main/java/com/rfs/common/DocumentReindexer.java

```diff
@@ -56,4 +76,31 @@ private String convertToBulkRequestBody(List<String> bulkSections) {
         }
         return builder.toString();
     }
+
+    private static java.util.function.Predicate<String> bufferPredicate(int maxItems, long maxSizeInBytes) {
```

Comment:
can you rename this to something more like statefulSizeSegmentingPredicate or anything more descriptive.  I'd much rather see the cap integrated with the collecting logic directly.

Reply:
Thanks!